File: 백준/Silver/11724. 연결 요소의 개수/연결 요소의 개수.py
from collections import deque
import sys
input = sys.stdin.readline

def bfs(graph, start, visited):
    q = deque([start])
    visited[start] = True
    while q:
        now = q.popleft()
        for i in graph[now]:
            if not visited[i]:
                q.append(i)
                visited[i] = True

n, m = map(int, input().split())
graph = [[]for _ in range(n+1)]
visited = [0] * (n+1)
for _ in range(m):
    a, b = map(int, input().split())
    graph[a].append(b)
    graph[b].append(a)

cnt = 0
for i in range(1, n + 1):
    if not visited[i]:
        bfs(graph, i, visited)
        cnt += 1
print(cnt)

File: 백준/Silver/11724. 연결 요소의 개수/연결 요소의 개수.kt
fun main() {
    val (n, m) = readLine()!!.split(" ").map { it.toInt() }
    val graph = Array(n + 1) { mutableListOf<Int>() }
    val visited = BooleanArray(n + 1) { false }

    repeat(m) {
        val (u, v) = readLine()!!.split(" ").map { it.toInt() }
        graph[u].add(v)
        graph[v].add(u)
    }

    fun dfs(v: Int) {
        visited[v] = true
        for (u in graph[v]) {
            if (!visited[u]) {
                dfs(u)
            }
        }
    }
    var components = 0
    for (i in 1..n) {
        if (!visited[i]) {
            dfs(i)
            components++
        }
    }

    println(components)
}